service/task.go

```diff
@@ -875,13 +875,17 @@ func (uis *UIServer) testLog(w http.ResponseWriter, r *http.Request) {
 		}()
 	}
 
-	template := "task_log.html"
 	if raw {
-		template = "task_log_raw.html"
-		uis.renderText.Stream(w, http.StatusOK, data, "base", template)
-	} else {
-		uis.render.WriteResponse(w, http.StatusOK, data, "base", template)
+		if logReader != nil {
+			gimlet.WriteText(w, logReader)
+		} else {
+			uis.renderText.Stream(w, http.StatusOK, data, "base", "task_log_raw.html")
+		}
+		return
 	}
+
+	go apimodels.ReadBuildloggerToChan(r.Context(), taskID, logReader, data.Buildlogger)
```

Comment:
I know we don't do this for the task log and that gimlet routes already catch panics, but should we bother handling panics from this specially at all? It seems like ReadBuildloggerToChan could panic in a few places.

Reply:
Added a panic handler, good thing you mentioned this cause this code was actually causing a panic! Caught it with the splunk logs in staging.